### ISSUE TITLE
Getting false positives

### ISSUE THREAD

**Issue by fent** [ssivanatarajan/stylelint-group-selectors#4]
Some false positive scenarios I've noticed

- in scss, when nesting, blocks that have no immediate properties are considered to have the same properties

```scss
.a {
  .b {
    color: blue;
  }
}

.c { // <- will get a warning here saying it can be merged with `.a`
  .d {
    font-size:  12pt;
  }
}
```

- media queries are ignored

```css
@media (max-width: 100px) {
  .a {
    color: blue;
  }
}

@media (max-width: 200px) {
  .b { // <-- warning here saying it can be merged with `.a` above despite different @media query
    color: blue;
  }
}
```

- nesting in scss is ignored
```scss
.a {
  background: green;
  .b {
    color: red;
  }
}
.c { // <-- warning here saying it can be merged with `.b`
  color: red;
}
```
although the above warning is true, you may not always want to pull `.b` out of that block since `.a` could be a complicated selector that you may not want to repeat typing, and grouping `.b` inside `.a` can provide additional context.

**Comment by jigneshchudasama** [comment]
Any update on this?

**Comment by ssivanatarajan** [comment]
@fent unable to reproduce the media query case. 
The nested selector case  in SCSS was fixed

**Comment by fent** [comment]
hmm, i haven't used this in a while. i noticed that the test file only tests one case https://github.com/ssivanatarajan/stylelint-group-selectors/blob/master/_tests_/index.js


> The nested selector case in SCSS was fixed

that's good!

**Comment by ssivanatarajan** [comment]
@fent  Will add more test cases
